File: Proj_NSURLProtocol/Pro_NSURLProtocol/Pro_NSURLProtocol/ViewController.m
//
//  ViewController.m
//  Pro_NSURLProtocol
//
//  Created by 申政 on 2022/4/28.
//

#import "ViewController.h"
#import <WebKit/WebKit.h>

@interface ViewController ()
@property (nonatomic, strong) WKWebView *webview;
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.webview = [[WKWebView alloc] initWithFrame:self.view.bounds];
    [self.webview setAutoresizingMask:NSViewWidthSizable | NSViewHeightSizable];
    [self.view addSubview:self.webview];
    
//  Test
//    NSURL *url = [NSURL URLWithString:@"http://raywenderlich.com"];
//    [self.webview loadRequest:[NSURLRequest requestWithURL:url]];
    
    NSString *localHtmlFilePath = [[NSBundle mainBundle] pathForResource:@"index" ofType:@"html"];
    if ([localHtmlFilePath length]) {
        NSURL *url = [NSURL fileURLWithPath:localHtmlFilePath];
        [self.webview loadFileURL:url allowingReadAccessToURL:[url URLByDeletingLastPathComponent]];
    } else {
        NSLog(@"Local file path not exist!");
    }
}

@end

File: Pro_URLSchemeTask/Pro_URLSchemeTask/ViewController.swift
//
//  ViewController.swift
//  Pro_URLSchemeTask
//
//  Created by 申政 on 2022/5/5.
//

import Cocoa
import WebKit

class ViewController: NSViewController {

    var webView: WKWebView?
    
    var urlSchemeHandler: MyURLSchemeHandler!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        urlSchemeHandler = MyURLSchemeHandler()
        let config = WKWebViewConfiguration()
        config.setURLSchemeHandler(urlSchemeHandler, forURLScheme: "myapp")
        
        webView = WKWebView(frame: view.bounds, configuration: config)
        webView?.autoresizingMask = [.width, .height]
        view.addSubview(webView!)
        //webView?.load(URLRequest(url: URL(string: "https://www.baidu.com")!))
        let path = Bundle.main.path(forResource:"index", ofType: "html")
        if let path = path {
            let url = URL(fileURLWithPath: path)
            webView?.load(URLRequest(url: url))
        }
    }
    
}
